hollow/src/main/java/com/netflix/hollow/api/producer/validation/RecordCountVarianceValidator.java

```diff
@@ -52,33 +52,71 @@ public RecordCountVarianceValidator(String typeName, float allowableVariancePerc
 		
-		// TODO: log message indicating previous state is 0. Can happen for new name space. And also can happen 
-		// when a type gets to zero count then the validation will be skipped.  
-		if(previousCardinality == 0){
-			log.log(Level.WARNING, "previousCardinality is 0. Not running RecordCountVarianceValidator for type "+typeName
-					+". This scenario is not expected except when starting a new namespace.O");
+		if(previousCardinality  == 0){
+			handleEndValidation(builder, Status.SKIP, false, String.format(ZERO_PREVIOUS_COUNT_WARN_MSG_FORMAT, typeName));
 			return;
 		}
 
-		float actualChangePercent = getChangePercent(latestCardinality, previousCardinality);
-		if (Float.compare(actualChangePercent, allowableVariancePercent) > 0) {
-			throw new ValidationException("RecordCountVarianceValidator for type " + typeName
-					+ " failed. Actual variance: " + actualChangePercent + "%; Allowed variance: "
-					+ allowableVariancePercent + "%;  previous cycle record count: " + previousCardinality
-							+ "; current cycle record count: " + latestCardinality);
-					
+		float actualChangePercent = getChangePercent(latestCardinality , previousCardinality );
+		builder.addAdditionalInfo(ACTUAL_CHANGE_PERCENT_NAME, String.valueOf(actualChangePercent));
+		
+		if (Float.compare(actualChangePercent , allowableVariancePercent) > 0) {
+			String message = String.format(FAILED_RECORD_COUNT_VALIDATION, typeName, actualChangePercent, allowableVariancePercent);
+			handleEndValidation(builder, Status.FAIL, true, message);
 		}
+		handleEndValidation(builder, Status.SUCCESS, false, null);
+	}
+	
+	@Override
+	public String toString(){
+		if(status != null)
+			return status.toString();
+		return("RecordCountVarianceValidator status for "+typeName+" is null. This is unexpected. Please check validator definition.");
+	}
+
+	private Builder initializeForValidation(ReadState readState) {
+		status = null;
+		Builder builder = ValidatorStatus.builder().withVersion(readState.getVersion());
+		builder.addAdditionalInfo(ALLOWABLE_VARIANCE_PERCENT_NAME, String.valueOf(allowableVariancePercent));
+		builder.addAdditionalInfo(DATA_TYPE_NAME, typeName);
+		return builder;
 	}
 
 	float getChangePercent(int latestCardinality, int previousCardinality) {
 		int diff = Math.abs(latestCardinality - previousCardinality);
 		float changePercent = ((float)100.0* diff)/(float)previousCardinality;
 		return changePercent;
 	}
+	
+	private void handleEndValidation(Builder builder, Status status, boolean throwException, String message){
+		ValidationException validationException = null;
+		builder.withStatus(status);
+		builder.withMessage(message);
+		if(throwException){
+			validationException = new ValidationException(message);
+			builder.withThrowable(validationException);
+			this.status = builder.build();
+			throw validationException;
+		}
+		this.status = builder.build();
+	}
+	
+	private static final String DATA_TYPE_NAME = "Typename";
```

Comment:
Minor: but should this be at the top of the class?   Don't recall seeing class fields defined at the bottom of class before so thought it might be good to be consistent.

Reply:
Typically they are defined at the top and I am not sure why they are defined at the top. I feel the definitions clutter the top and distract from the class itself and so moved it down.